File: viewer/lib/models/global_partner.dart
/// Parceiro global exibido no banner da Central.
class GlobalPartner {
  final String id;
  final String name;
  final String? description;
  final String? logoUrl;
  final String? offerText;
  final String? externalUrl;
  final String? buttonLabel;
  final int? featuredOrder;
  final bool isActive;

  const GlobalPartner({
    required this.id,
    required this.name,
    this.description,
    this.logoUrl,
    this.offerText,
    this.externalUrl,
    this.buttonLabel,
    this.featuredOrder,
    this.isActive = true,
  });

  factory GlobalPartner.fromJson(Map<String, dynamic> json) {
    return GlobalPartner(
      id: json['id'] as String,
      name: json['name'] as String,
      description: json['description'] as String?,
      logoUrl: json['logo_url'] as String?,
      offerText: json['offer_text'] as String?,
      externalUrl: json['external_url'] as String?,
      buttonLabel: json['button_label'] as String?,
      featuredOrder: json['featured_order'] as int?,
      isActive: json['is_active'] as bool? ?? true,
    );
  }
}

File: app/models/global_partner.py
"""Parceiro global exibido no banner da Central para todos os alunos."""
from __future__ import annotations

from sqlalchemy import Boolean, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base
from app.models.base import UUIDMixin


class GlobalPartner(Base, UUIDMixin):
    __tablename__ = "global_partners"

    name: Mapped[str] = mapped_column(String(255), nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    logo_url: Mapped[str | None] = mapped_column(String(512), nullable=True)
    offer_text: Mapped[str | None] = mapped_column(Text, nullable=True)
    external_url: Mapped[str | None] = mapped_column(String(512), nullable=True)
    featured_order: Mapped[int | None] = mapped_column(Integer, nullable=True)
    is_active: Mapped[bool] = mapped_column(
        Boolean,
        nullable=False,
        server_default="true",
    )
